SourceCode/h3/image_colored_by_norm_vectors.cpp

```diff
@@ -0,0 +1,83 @@
+#include <fstream>
+#include <cmath>
+
+// Output image resolution
+static const int imageWidth = 6000;
+static const int imageHeight = 2000;
+static const int maxColorComponent = 255;
+
+class Vector {
+    float x, y, z;
+public:
+    Vector(float x, float y, float z) {
+        this->x = x;
+        this->y = y;
+        this->z = z;
+    }
+
+    float length() const {
+        return sqrt(x * x + y * y + z * z);
+    }
+    void normalize() {
+        x = x / length();
```

Comment:
Instead of calling length() 3 times you could cache its result, this way you'll use sqrt only once when normalizing. The compiler could do that for us, but it's better to be sure.

Reply:
That's a good point. Will implement.